Lib/string.py

```diff
@@ -89,6 +66,25 @@ class Template(metaclass=_TemplateMetaclass):
     braceidpattern = None
     flags = _re.IGNORECASE
 
+    def __init_subclass__(cls):
```

Comment:
This seems like a better approach, but if you're modernizing this then I have a few other suggestions.